presto-spi/src/main/java/com/facebook/presto/spi/relation/DomainTranslator.java

```diff
@@ -55,6 +56,21 @@ public ExtractionResult(TupleDomain<T> tupleDomain, RowExpression remainingExpre
             this.remainingExpression = requireNonNull(remainingExpression, "remainingExpression is null");
         }
 
+        public ExtractionResult intersect(ExtractionResult other)
```

Comment:
Could you move this function inside HiveMetadata? Let's make the spi light-weighted.